onnx/shape_inference/implementation.cc

```diff
@@ -493,7 +502,8 @@ void InferShapes(
     const bool check_type,
     const std::string& save_path,
     const ISchemaRegistry* schema_registry,
-    const int error_mode) {
+    const int error_mode,
+    const int data_prop) {
```

Comment:
making this bool makes more sense... something like bool enable_data_prop
reason for error_mode being int is tomorrow we can introduce more modes easily but data propagation is binary yes or no